src/shared/defaultAwsContextCommands.ts

```diff
@@ -186,9 +186,8 @@ export class DefaultAWSContextCommands {
 
             return await this.promptAndCreateNewCredentialsFile()
         } else {
-            const profileNames = (
-                await CredentialsProviderManager.getInstance().getAllCredentialsProviders()
-            ).map(provider => asString(provider.getCredentialsProviderId()))
+            const profiles = await CredentialsProviderManager.getInstance().getProfiles()
+            const profileNames = Object.keys(profiles)
```

Comment:
This pattern is repeated in this PR. What if CredentialsProviderManager had a `getAllCredentialsProviderIds` method instead?

Reply:
Adding special methods like `getAllCredentialsProviderIds` that do roughly the same thing as idioms like `Object.keys` is strictly a regression. It adds new names that must be understood by other developers, for little gain.

Reply:
That's reasonable. I found it interesting that you introduced function that produced a map, only to pull the keys off it, which prompted the comment.